From 69487945b22485f7f240a00b4135d479cd75e0bc Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Mon, 5 Nov 2018 16:57:34 +0800
Subject: [PATCH 5025/5242] MLK-20196-1 i2c: xen-i2cback: fix num_msg usage

commit  7f84a84d3f5d1d47283fac9b42d7652e6e39a252 from
https://source.codeaurora.org/external/imx/linux-imx.git

When req.num_msg is less than 2, also need initialze num_msg.
Also modify the if/else if to use num_msg

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/i2c/busses/xen-i2cback.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/xen-i2cback.c b/drivers/i2c/busses/xen-i2cback.c
index fd5f832..ccc67c7 100644
--- a/drivers/i2c/busses/xen-i2cback.c
+++ b/drivers/i2c/busses/xen-i2cback.c
@@ -94,8 +94,10 @@ static bool i2cback_handle_int(struct i2cback_info *info)
 		allow_access = i2cback_access_allowed(info, &req);
 		if (allow_access) {
 			/* Write/Read sequence */
-			if (req.num_msg > I2CIF_MAX_MSG)
+			num_msg = req.num_msg;
+			if (num_msg > I2CIF_MAX_MSG)
 				num_msg = I2CIF_MAX_MSG;
+
 			for (i = 0; i < num_msg; i++) {
 				msg[i].addr = req.msg[i].addr;
 				msg[i].len = req.msg[i].len;
@@ -130,7 +132,7 @@ static bool i2cback_handle_int(struct i2cback_info *info)
 				memcpy(tmp_buf, req.write_buf, I2CIF_BUF_LEN);
 				ret = i2c_transfer(info->adapter, msg,
 						   num_msg);
-			} else if (req.num_msg == 1) {
+			} else if (num_msg == 1) {
 				msg[0].buf = tmp_buf;
 				if (!(msg[0].flags & I2C_M_RD))
 					memcpy(tmp_buf, req.write_buf,
-- 
1.7.9.5

